File: Cryptography/monoAlphabeticCipherRandom.go
package main

import (
	"fmt"
	"time"
)

func getRandomTable() map[rune]rune {
	subTable := map[rune]rune{}
	allowedLetters := map[rune]bool{}

	for i := 'n'; i <= 'z'; i++ {
		allowedLetters[i] = true
	}

	for i := 'a'; i <= 'm'; i++ {
		for {
			var val rune = rune('n' + (time.Now().UnixMilli() % 13))
			if allowedLetters[val] == true {
				subTable[i] = val
				allowedLetters[val] = false
				break
			}
		}

	}
	return subTable
}

func macRandom(subTable map[rune]rune, text string) string {
	result := ""
	for _, letter := range text {
		value, found := subTable[letter]
		if found {
			result += string(value)
		} else {
			for k, v := range subTable {
				if v == letter {
					result += string(k)

				}
			}
		}

	}

	return result
}

func task4() {
	var input string
	fmt.Println("Enter Text to be encrypted:")
	fmt.Scanf("%s", &input)
	subTable := getRandomTable()
	encrypted := macRandom(subTable, input)
	decrypted := macRandom(subTable, encrypted)
	fmt.Println("Encrypted: " + encrypted)
	fmt.Println("Decrypted: " + decrypted)
}

File: Cryptography/wasif-sir/monoAlphabeticCipherRandom.cpp
#include <bits/stdc++.h>
using namespace std;
unordered_map<char, char> getRandomTable() {
    unordered_map<char, char> subTable;
    unordered_map<char, bool> allowedLetters;
    for (char i = 'n'; i <= 'z'; i++) {
        allowedLetters[i] = true;
    }
    for (char i = 'a'; i <= 'm'; i++) {
        while (true) {
            long nowInMilli =
            chrono::duration_cast<chrono::milliseconds>(chrono::system_clock::now().time_since_epoch()).count();
            char val = 'n' + nowInMilli%13;
            if (allowedLetters[val] == true) {
                subTable[i] = val;
                allowedLetters[val] = false;
                break;
            }
        }
    }
    return subTable;
}
string macRandom(unordered_map<char, char>& subTable, string text) {
    string result = "";
    for (char letter : text) {
        auto found = subTable.find(letter);
        if (found != subTable.end()) {
            result += found->second;
        } else {
            
            for (auto& pair : subTable) {
                if (pair.second == letter) {
                    result += pair.first;
                    break;
                }
            }
        }
    }
    return result;
}
int main() {
    string input;
    cout << "Enter Text to be encrypted:" << endl;
    cin >> input;
    unordered_map<char, char> subTable = getRandomTable();
    string encrypted = macRandom(subTable, input);
    string decrypted = macRandom(subTable, encrypted);
    cout << "Encrypted: " << encrypted << endl;
    cout << "Decrypted: " << decrypted << endl;
    return 0;
}